drivers/comparator/comparator_stm32_comp.c

```diff
@@ -0,0 +1,376 @@
+
+#if defined(SOC_SERIES_STM32H7X)
+#define STM32_COMP_DT_INST_COMP_CONFIG_INIT(inst)                                                  \
+	{                                                                                          \
+		.PowerMode = STM32_COMP_DT_POWER_MODE(inst),                                       \
+		.InputPlus = STM32_COMP_DT_INST_P_IN(inst),                                        \
+		.InputMinus = STM32_COMP_DT_INST_N_IN(inst),                                       \
+		.InputHysteresis = STM32_COMP_DT_INST_HYST_MODE(inst),                             \
+		.OutputPolarity = STM32_COMP_DT_INST_INV_OUT(inst),                                \
+		.OutputBlankingSource = STM32_COMP_DT_INST_BLANK_SEL(inst),                        \
+	}
+#else /* SOC_SERIES_STM32H7X */
+#define STM32_COMP_DT_INST_COMP_CONFIG_INIT(inst)                                                  \
+	{                                                                                          \
+		.InputPlus = STM32_COMP_DT_INST_P_IN(inst),                                        \
+		.InputMinus = STM32_COMP_DT_INST_N_IN(inst),                                       \
+		.InputHysteresis = STM32_COMP_DT_INST_HYST_MODE(inst),                             \
+		.OutputPolarity = STM32_COMP_DT_INST_INV_OUT(inst),                                \
+		.OutputBlankingSource = STM32_COMP_DT_INST_BLANK_SEL(inst),                        \
+	}
+#endif /* SOC_SERIES_STM32H7X */
+
+struct stm32_comp_config {
+	COMP_TypeDef *comp;
+	const struct stm32_pclken *pclken;
+	const struct pinctrl_dev_config *pincfg;
+	void (*irq_init)(void);
+	const uint32_t irq_nr;
+	const LL_COMP_InitTypeDef comp_config;
+	const uint32_t exti_line;
+	const uint8_t exti_line_number;
+	const bool lock_enable;
+	const bool miller_effect_hold_enable;
+};
+
+struct stm32_comp_data {
+	comparator_callback_t callback;
+	void *user_data;
+};
+
+__maybe_unused static bool stm32_comp_is_resumed(void)
+{
+#if CONFIG_PM_DEVICE
+	enum pm_device_state state;
+
+	(void)pm_device_state_get(DEVICE_DT_INST_GET(0), &state);
+	return state == PM_DEVICE_STATE_ACTIVE;
+#else
+	return true;
+#endif /* CONFIG_PM_DEVICE */
+}
+
+static int stm32_comp_get_output(const struct device *dev)
+{
+	const struct stm32_comp_config *cfg = dev->config;
+
+	return LL_COMP_ReadOutputLevel(cfg->comp);
+}
+
+static int stm32_comp_set_trigger(const struct device *dev, enum comparator_trigger trigger)
+{
+	const struct stm32_comp_config *cfg = dev->config;
+	struct stm32_comp_data *data = dev->data;
+	int ret = 0;
+
+	LL_EXTI_InitTypeDef exti_init_data = {0};
+
+	LL_EXTI_StructInit(&exti_init_data);
+	exti_init_data.LineCommand = ENABLE;
+	exti_init_data.Mode = LL_EXTI_MODE_IT;
+
+	if (cfg->exti_line_number < 32U) {
+		exti_init_data.Line_0_31 = cfg->exti_line;
+	} else if (cfg->exti_line_number < 64U) {
+		exti_init_data.Line_32_63 = cfg->exti_line;
+	} else {
+		exti_init_data.Line_64_95 = cfg->exti_line;
```

Comment:
For the whole exti usage, did you consider using `drivers/interrupt_controller/intc_exti_stm32.c`.
I haven't dig deep into the implementation, but I'd like to avoid code duplication and potential resource usage conflicts. See stm32 gpio driver for example on how this is used.
It is possible that some adaptation would be required to exti driver if needed.

I'll need to dig deeper on that point but that would be the main blocker to me untill we're sure this is the best option.

Reply:
No. I didn't. But it makes sense to reuse of course.  I've skimmed through `drivers/interrupt_controller/intc_exti_stm32.c` and see a lot of simplifications. For example only EXTI lines 0 to 31 will be considered, since it was primarily developed for GPIO and gpio uses lines 0 to 15. 
My verdict: I would need a deep rework of this driver, but it shall be possible. 
Although I need support regarding diversity of st's exti implementations. At the moment I am only familiar with h74xx and g4xx. 
Does `drivers/interrupt_controller/intc_exti_stm32.c` also has tests implemented?

Reply:
@erwango Would you create an issue? I can then reimplement, I think. Or do you have anybody else already working on this topic?

Reply:
> @erwango Would you create an issue? I can then reimplement, I think. Or do you have anyone else already working on this topic?

I'll create the issue. This was not in our short term plans as you just created the need ;)